File: app/src/main/java/com/hfad/myferma/WarehouseFragment.java
package com.hfad.myferma;

import android.app.FragmentTransaction;
import android.database.Cursor;
import android.os.Bundle;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.hfad.myferma.WriteOff.WriteOffFragment;
import com.hfad.myferma.db.MyConstanta;
import com.hfad.myferma.db.MyFermaDatabaseHelper;
import com.hfad.myferma.db.MydbManagerMetod;
import com.hfad.myferma.incubator.AddIncubatorFragment;
import com.hfad.myferma.incubator.IncubatorMenuFragment;
import com.hfad.myferma.incubator.ListAdapterIncubator;

import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;


public class WarehouseFragment extends Fragment implements View.OnClickListener {

    private MyFermaDatabaseHelper myDB;
    private RecyclerView recyclerView;
    private List<String> productList;
    private String unit = null;
    private DecimalFormat f, eggFormat;

    //Создание фрагмента
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Открываем БД
        myDB = new MyFermaDatabaseHelper(getActivity());

        View layout = inflater.inflate(R.layout.fragment_warehouse, container, false);

        // Назначаем кнопки
        Button addSale = (Button) layout.findViewById(R.id.writeOff_button);
        addSale.setOnClickListener(this);

        Button incubator = (Button) layout.findViewById(R.id.incubator_button);
        incubator.setOnClickListener(this);

        //Настройка листа
        productList = new ArrayList();
        add();

        // Настраиваем адаптер
        recyclerView = layout.findViewById(R.id.recyclerView);

        ProductAdapter productAdapter = new ProductAdapter(add1(),"", " Шт.");
        recyclerView.setAdapter(productAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));

        f = new DecimalFormat("0.00");
        eggFormat = new DecimalFormat("0");

        return layout;
    }

    public void add() {
        Cursor cursor = myDB.readAllDataProduct();

        while (cursor.moveToNext()) {
            String product = cursor.getString(1);
            productList.add(product);
        }
        cursor.close();
    }


    public Map add1() {
        Map<String, Double> tempList = new HashMap<>();

        for (String product : productList) {

            Cursor cursor = myDB.idProduct1(MyConstanta.TABLE_NAME, MyConstanta.TITLE, product);

            if (cursor != null && cursor.getCount() != 0) {

                while (cursor.moveToNext()) {
                    Double productUnit = cursor.getDouble(2);
                    if (tempList.get(product) == null) {
                        tempList.put(product, productUnit);
                    } else {
                        double sum = tempList.get(product) + productUnit;
                        tempList.put(product, sum);
                    }
                }
                cursor.close();

                Cursor cursorSale = myDB.idProduct1(MyConstanta.TABLE_NAMESALE, MyConstanta.TITLESale, product);

                while (cursorSale.moveToNext()) {
                    Double productUnit = cursorSale.getDouble(2);
                    double minus = tempList.get(product) - productUnit;
                    tempList.put(product, minus);
                }
                cursorSale.close();

                Cursor cursorWriteOff = myDB.idProduct1(MyConstanta.TABLE_NAMEWRITEOFF, MyConstanta.TITLEWRITEOFF, product);

                while (cursorWriteOff.moveToNext()) {
                    Double productUnit = cursorWriteOff.getDouble(2);
                    double minusWriteOff = tempList.get(product) - productUnit;
                    tempList.put(product, minusWriteOff);
                }
                cursorWriteOff.close();
            } else {
                tempList.put(product, 0.0);
            }
        }

        return tempList;
    }

    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.writeOff_button:
                onClickButtonOff(v, new WriteOffFragment());
                break;
            case R.id.incubator_button:
                onClickButtonOff(v, new IncubatorMenuFragment());
                break;
        }
    }

    public void onClickButtonOff(View view, Fragment fragment) {

        getActivity().getSupportFragmentManager().beginTransaction()
                .replace(R.id.conteiner, fragment, "visible_fragment")
                .addToBackStack(null)
                .commit();

//        ft.replace(R.id.conteiner, fragment, "visible_fragment");
//        ft.addToBackStack(null);
//        ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE);
//        ft.commit();
    }

}

File: app/src/main/java/com/hfad/myferma/WarehouseFragment.kt
package com.hfad.myferma

import android.database.Cursor
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.hfad.myferma.menu.WriteOffFragment
import com.hfad.myferma.db.MyConstanta
import com.hfad.myferma.db.MyFermaDatabaseHelper
import com.hfad.myferma.incubator.MenuIncubators.IncubatorMenuFragment
import java.text.DecimalFormat


class WarehouseFragment : Fragment() {
    private lateinit var myDB: MyFermaDatabaseHelper
    private var productList = mutableListOf<String>()
    private val unit: String? = null
    private var f = DecimalFormat("0.00")
    private var eggFormat = DecimalFormat("0")

    //Создание фрагмента
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val layout: View = inflater.inflate(R.layout.fragment_warehouse, container, false)
        // Открываем БД
        myDB = MyFermaDatabaseHelper(requireContext())

        // Назначаем кнопки
        val writeOffFragment: Button = layout.findViewById<View>(R.id.writeOff_button) as Button
        writeOffFragment.setOnClickListener { onClickButtonOff(WriteOffFragment()) }

        val incubator: Button = layout.findViewById<View>(R.id.incubator_button) as Button
        incubator.setOnClickListener { onClickButtonOff(IncubatorMenuFragment()) }

        //Настройка листа
        addArray()

        // Настраиваем адаптер
        val recyclerView = layout.findViewById<RecyclerView>(R.id.recyclerView)
        val productAdapter = ProductAdapter(add(), "", " Шт.")
        recyclerView.adapter = productAdapter
        recyclerView.layoutManager = LinearLayoutManager(activity)
        return layout
    }

    private fun addArray() {
        val cursor: Cursor = myDB.readAllDataProduct()
        while (cursor.moveToNext()) {
            val product: String = cursor.getString(1)
            productList.add(product)
        }
        cursor.close()
    }

    private fun add(): Map<String, Double> {

        val tempList: MutableMap<String, Double> = HashMap()

        for (product: String in productList) {

            val cursor = myDB.selectTableNameAndSumCount(
                MyConstanta.TABLE_NAME,
                MyConstanta.TITLE,
                product,
                MyConstanta.DISCROTION
            )

            if (cursor.count != 0) {

                cursor.moveToNext()
                tempList[product] = cursor.getDouble(1)
                cursor.close()

                val cursorSale =
                    myDB.selectTableNameAndSumCount(
                        MyConstanta.TABLE_NAMESALE,
                        MyConstanta.TITLESale,
                        product,
                        MyConstanta.DISCROTIONSale
                    )
                if (cursorSale.count != 0) {
                    cursorSale.moveToNext()
                    tempList[product] = tempList[product]!! - cursorSale.getDouble(1)
                }
                cursorSale.close()

                val cursorWriteOff = myDB.selectTableNameAndSumCount(
                    MyConstanta.TABLE_NAMEWRITEOFF,
                    MyConstanta.TITLEWRITEOFF,
                    product, MyConstanta.DISCROTIONSWRITEOFF
                )

                if (cursorWriteOff.count != 0) {
                    cursorWriteOff.moveToNext()
                    tempList[product] = tempList[product]!! - cursorWriteOff.getDouble(1)
                }
                cursorWriteOff.close()

            } else {
                tempList[product] = 0.0
            }

        }
        return tempList
    }

    private fun onClickButtonOff(fragment: Fragment) {
        requireActivity().supportFragmentManager.beginTransaction()
            .replace(R.id.conteiner, fragment, "visible_fragment")
            .addToBackStack(null)
            .commit()
    }
}